docs/data/data-grid/selection/selection.md

```diff
@@ -84,25 +84,89 @@ The following demo shows the prop in action:
+The `DataGridPremium` comes with the ability to select rows enabled by default.
+You can also select cells by turning on the `cellSelection` prop.
 
-:::warning
-Only use this API as the last option. Give preference to the props to control the grid.
+```tsx
+<DataGridPremium cellSelection />
+```
+
+To select a single cell, click on it or, with the cell focused, press <kbd><kbd class="key">Shift</kbd>+<kbd class="key">Space</kbd></kbd>.
+Multiple cells can be selected by holding <kbd class="key">Ctrl</kbd> while clicking the cells.
+A cell can be deselected by also clicking it while <kbd class="key">Ctrl</kbd> is pressed.
+
+To select all cells within a range, you can use one of the shortcuts available:
+
+- Click on a cell, then drag the mouse over another cell and release it
+
+- Click on a cell and, while holding <kbd class="key">Shift</kbd>, click on another cell—if a third cell is clicked the selection will restart from the last clicked cell
+
+- Using the arrow keys, focus on a cell, then hold <kbd class="key">Shift</kbd> and navigate to another cell—if <kbd class="key">Shift</kbd> is released and pressed again, the selection will restart from the last focused cell
+
+The following demo allows to explore the cell selection feature.
+It has row selection disabled initially, but both features can be enabled at the same time.
+
+{{"demo": "CellSelectionGrid.js", "bg": "inline"}}
+
+### Controlled cell selection
+
+You can control which cells are selected with the `cellSelectionModel` prop.
+This props accepts an object whose keys are the row IDs that contain selected cells.
+The value of each key is another object, where the keys correpond to the `field` of the selected cell and the value is `true` to select the cell, or `false` to deselect it.
+Removing the field from the object also deselects the cell.
+
+```tsx
+// Selects the cell with field=name from row with id=1
+<DataGrid
+  cellSelectionModel={{ 1: { name: true } }}
+/>
+
+// Unselects the cell with field=name from row with id=1
+<DataGrid
+  cellSelectionModel={{ 1: { name: false } }}
+/>
+```
+
+When a new selection is made, the callback passed to the `onCellSelectionModelChange` prop is called with the updated model.
+Use this value to update the current model.
+
+The following demo shows how these props can be combined to create an Excel-like formula field.
+
+{{"demo": "CellSelectionFormulaField.js", "bg": "inline"}}
+
+### Customize range styling
+
+When multiple selected cells make a range, specific class names are applied to the cells at the edges of this range.
+The class names applied are the following:
+
+- `MuiDataGrid-cell--rangeTop`: applied to all cells of the first row of the range
+
+- `MuiDataGrid-cell--rangeBottom`: applied to all cells of the last row of the range
+
+- `MuiDataGrid-cell--rangeLeft`: applied to all cells of the first column of the range
+
+- `MuiDataGrid-cell--rangeRight`: applied to all cells of the last column of the range
```

Comment:
Same feedback as above?

```suggestion
- `MuiDataGrid-cell--rangeTop`: applied to all cells of the first row of the range
- `MuiDataGrid-cell--rangeBottom`: applied to all cells of the last row of the range
- `MuiDataGrid-cell--rangeLeft`: applied to all cells of the first column of the range
- `MuiDataGrid-cell--rangeRight`: applied to all cells of the last column of the range
```

---

**Off-topic.** I would be in favor of going to http://github.com/mui/material-ui and adding a bit of margin-bottom for `ul > li`. e.g. 2px so it feels better. Added the idea to https://www.notion.so/mui-org/Opportunities-408a9771556a4715830d18f3749af753#66fe1b01bb3b495388492d2121674f00